mindsdb/interfaces/skills/sql_agent.py

```diff
@@ -119,13 +119,19 @@ def _resolve_table_names(self, table_names: List[str], all_tables: List[Identifi
             # by path
             tables_idx[tuple(table.parts)] = table
 
+        # Some LLMs (e.g. gpt-4o) may include backticks or quotes when invoking tools.
+        strip_chars = ' "\'\n\r'
+        # Remove backticks only if the actual table names don't have them.
+        # Querying data sources like MS OneDrive and Amazon S3 requires tables (files) to be wrapped in backticks.
+        if not any('`' in table_name.get_string() for table_name in all_tables):
+            strip_chars = strip_chars + '`'
+
         tables = []
         for table_name in table_names:
             if not table_name.strip():
                 continue
 
-            # Some LLMs (e.g. gpt-4o) may include backticks or quotes when invoking tools.
-            table_name = table_name.strip(' `"\'\n\r')
+            table_name = table_name.strip(strip_chars)
```

Comment:
Identifier have to be able to digest backticks and use them when it is required:
```python
Identifier('df1.`iris.csv`').parts
>>> ['df1', 'iris.csv']

Identifier('df1.`iris`').to_string()
>>>'df1.iris'

Identifier('df1.`iris.csv`').to_string()
>>>'df1.`iris.csv`'
```

Maybe we can left this line like this? 
`table_name = table_name.strip(' "\'\n\r')`

Reply:
@ea-rus Agreed. I've pushed some changes.